src/BlockchainWriter/DAO.ts

```diff
@@ -17,10 +23,16 @@ export class DAO {
     this.blockchainWriterCollection.insertOne({
       ipfsDirectoryHash,
       txId: null,
+      transactionCreationDate: null,
     })
 
   readonly findTransactionlessEntry = () => this.blockchainWriterCollection.findOne({ txId: null })
 
-  readonly setTransactionId = (ipfsDirectoryHash: string, txId: string) =>
-    this.blockchainWriterCollection.updateOne({ ipfsDirectoryHash }, { $set: { txId } }, { upsert: true })
+  readonly findAndUpdateByIPFSDirectoryHash = async ({ ipfsDirectoryHash, ...updates }: Entry) => {
```

Comment:
Shouldn't this be named just `updateBy...` or did you mean to use [findAndModify](https://docs.mongodb.com/manual/reference/method/db.collection.findAndModify/)?

Reply:
`updateByIPFSDirectoryHash` works for me, I think mongo func names are starting to invade my brain a bit.